app/views/play/CampaignView.coffee

```diff
@@ -1146,15 +1150,14 @@ module.exports = class CampaignView extends RootView
       return false unless @classroom?   
       return @classroom.getSetting('xp')
 
-
-    if what in ['settings', 'leaderboard', 'back-to-campaigns', 'poll', 'items', 'heros', 'achievements', 'clans', 'poll', 'buy-gems']
+    if what in ['settings', 'leaderboard', 'back-to-campaigns', 'poll', 'items', 'heros', 'achievements', 'clans', 'poll']
       return !isStudent
 
     if what in ['back-to-classroom']
       return isStudent
 
-    if what in 'buy-gems'
-      return (me.get('anonymous') is false || me.get('iosIdentifierForVendor') || isIPadApp) && !me.freeOnly() 
+    if what in ['buy-gems', 'premium']
+      return not (me.get('iosIdentifierForVendor') || application.isIPadApp) && !me.freeOnly() && !isStudent
 
```

Comment:
Can you explain these changes to `'buy-gems'` behavior? I'm not sure why they're changed, or what the intended changes are.